polaris-core/src/test/java/org/apache/polaris/core/storage/PolarisStorageConfigurationInfoTest.java

```diff
@@ -0,0 +1,91 @@
+package org.apache.polaris.core.storage;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.stream.Stream;
+import org.apache.polaris.core.PolarisDefaultDiagServiceImpl;
+import org.apache.polaris.core.storage.aws.ImmutableAwsStorageConfigurationInfo;
+import org.apache.polaris.core.storage.azure.ImmutableAzureStorageConfigurationInfo;
+import org.apache.polaris.core.storage.gcp.ImmutableGcpStorageConfigurationInfo;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class PolarisStorageConfigurationInfoTest {
+
+  public static Stream<Arguments> serialize() {
+    return Stream.of(
+        Arguments.of(
+            ImmutableAwsStorageConfigurationInfo.builder()
+                .allowedLocations(
+                    List.of(
+                        "s3://bucket/path/to/warehouse",
+                        "s3://bucket/anotherpath/to/warehouse",
+                        "s3://bucket2/warehouse/"))
+                .roleARN("arn:aws:iam::012345678901:role/jdoe")
+                .externalId("externalId")
+                .userARN("userARN")
+                .build(),
+            "AwsStorageConfigurationInfo"),
+        Arguments.of(
+            ImmutableAzureStorageConfigurationInfo.builder()
+                .allowedLocations(
+                    List.of(
+                        "abfs://account@container.com/path/to/warehouse",
+                        "abfs://account@container.com/anotherpath/to/warehouse",
+                        "abfs://account2@container.com/warehouse/"))
+                .consentUrl("https://login.microsoftonline.com/tenantId/oauth2/authorize")
+                .multiTenantAppName("appName")
+                .tenantId("tenantId")
+                .build(),
+            "AzureStorageConfigurationInfo"),
+        Arguments.of(
+            ImmutableGcpStorageConfigurationInfo.builder()
+                .allowedLocations(
+                    List.of(
+                        "gs://bucket/path/to/warehouse",
+                        "gs://bucket/anotherpath/to/warehouse",
+                        "gs://bucket2/warehouse/"))
+                .gcpServiceAccount("serviceAccount")
+                .build(),
+            "GcpStorageConfigurationInfo"),
+        Arguments.of(
+            ImmutableFileStorageConfigurationInfo.builder()
+                .allowedLocations(
+                    List.of(
+                        "file:///path/to/warehouse",
+                        "file:///anotherpath/to/warehouse",
+                        "file:///warehouse/"))
+                .build(),
+            "FileStorageConfigurationInfo"));
+  }
+
+  @ParameterizedTest
+  @MethodSource
+  void serialize(PolarisStorageConfigurationInfo instance, String typeName) {
+    String json = instance.serialize();
+    assertThat(json).contains("\"@type\":\"" + typeName + "\"");
```

Comment:
@snazy I manually ran this test against `main` and verified that the serialized json is similar: the `@type` attribute is the same and the rest is similar in the sense that the new json from this branch generally contains more properties than the json from `main` (you cannot just add `@JsonIgnore` and expect the immutables library to honor it). Since the `ObjectMapper` is configured with `FAIL_ON_UNKNOWN_PROPERTIES` I think we are good.

Reply:
Sorry that's what I meant: `FAIL_ON_UNKNOWN_PROPERTIES` is disabled.

Reply:
Also looked at the serialized JSONs from `main` and this branch - and the result's equal, except for AWS (see above).